qiime2/sdk/usage.py

```diff
@@ -6,16 +6,38 @@
 # The full license is in the file LICENSE, distributed with this software.
 # ----------------------------------------------------------------------------
 
+"""
+The Usage module enables generating examples for multiple QIIME 2 interfaces.
+
+The Usage API provides an interface-agnostic way for QIIME 2 plugin developers
+to define examples of how to use their plugin’s actions. This enables the
+programmatic generation of examples for all QIIME 2 interfaces, eliminating the
+need to maintain specific examples for multiple interfaces.
+
+Examples
+--------
+Please see `qiime2.core.testing.examples` for examples.
+"""
```

Comment:
This docstring still ends up in the rendered dev docs because of the `.. automodule:: qiime2.sdk.usage` directive.

Moving it was prompted by running `numpydoc --validate` which reminded me about [numpy's advice for documenting modules](https://numpydoc.readthedocs.io/en/latest/format.html#documenting-modules).

Reply:
> This docstring still ends up in the rendered dev docs because of the .. automodule:: qiime2.sdk.usage directive.

Yep, I saw that change. My point is that this prose here is more appropriate as high-level plugin-developer-facing documentation, and less appropriate as a general-purpose module-level docstring.

I'm not saying we can't have a module-level docstring, I'm just saying that _this_ docstring is better for plugin-dev docs. As the docs grow, the differences between the plugin-dev docs vs the sdk docs will become more cumbersome, so let's split it from the outset.

Reply:
That makes sense. Thanks for clarifying.